File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-6.py
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        count = {}
        max_f = 0
        result = 0
        left = 0

        for right in range(len(s)):
            count[s[right]] = 1 + count.get(s[right], 0)

            max_f = max(max_f, count[s[right]])

            while (right - left + 1) - max_f > k:
                count[s[left]] -= 1
                left += 1
            
            result = max((right - left + 1), result)
        
        return result

File: Data Structures & Algorithms/count-good-nodes-in-binary-tree/submission-6.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    public int goodNodes(TreeNode root) {
        return countNodes(root, root.val);
    }

    public int countNodes(TreeNode root, int maxValue){
        if(root == null){
            return 0;
        }

        int result = (maxValue <= root.val) ? 1 : 0;
        maxValue = Math.max(maxValue, root.val);

        result += countNodes(root.left, maxValue);
        result += countNodes(root.right, maxValue);

        return result;
    }
}
